### PR TITLE
Addons docs

### PR DESCRIPTION
**Pull request by iamdustan**
@spicyj The second commit is questionable. I grepped for `react/addons`...not sure if I missed anything from how the docs are built though.

### CODE REVIEW

File: npm-react/README.md [hunk 1]
```diff
@@ -14,7 +14,9 @@ To use React in production mode, set the environment variable `NODE_ENV` to `pro
 ```js
 var React = require('react');
 
-// You can also access ReactWithAddons.
-var React = require('react/addons');
+// You can also access React's addons specifically.
```
Comment (zpao) [inline npm-react/README.md]: Might be good to say something like "See ___ for the list of available addons."

File: npm-react-codemod/transforms/utils/ReactUtils.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ module.exports = function(j) {
       .filter(j.filters.VariableDeclarator.requiresModule(module))
       .size() === 1;
 
+  // TODO: any specific addon?
```
Comment (zpao) [inline npm-react-codemod/transforms/utils/ReactUtils.js]: cc @cpojer who can probably answer this. I'm going to guess we don't want to change anything.

File: docs/docs/10-addons.md [hunk 3]
```diff
@@ -22,4 +22,5 @@ The add-ons below are in the development (unminified) version of React only:
 
 To get the add-ons, use `react-with-addons.js` (and its minified counterpart) rather than the common `react.js`.
```
Comment (zpao) [inline docs/docs/10-addons.md]: ReactWithAddons.js for the browser isn't going anywhere yet so we should probably make it clear that `React.addons.{addon}` exists there.

File: npm-react-codemod/test/pure-render-mixin-test2.output.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
-var React = require('react/addons');
+var React = require('react');
+var shallowCompare = require('react/addons/shallowCompare');
 
 var MyComponent = React.createClass({
   shouldComponentUpdate: function(nextProps, nextState) {
-    return React.addons.shallowCompare(this, nextProps, nextState);
```
Comment (spicyj) [inline npm-react-codemod/test/pure-render-mixin-test2.output.js]: This code is generated by https://github.com/facebook/react/blob/master/npm-react-codemod/transforms/pure-render-mixin.js. If you're going to change these, can you change that source too and make sure the tests pass?

File: docs/downloads.md [hunk 5]
```diff
@@ -74,7 +74,7 @@ var React = require('react');
 React.render(...);
 ```
 
-If you'd like to use any [add-ons](/react/docs/addons.html), use `var React = require('react/addons');` instead.
+If you'd like to use any [add-ons](/react/docs/addons.html), use `var React = require('react/addons/{addon}');` instead.
```
Comment (spicyj) [inline docs/downloads.md]: Not `var React =`.

File: npm-react-codemod/test/pure-render-mixin-test2.output.js (hunk 4 above)
Comment (cpojer) [inline npm-react-codemod/test/pure-render-mixin-test2.output.js]: yeah, this actually requires changing the codemod script as well to output the proper require statement.

Then in `npm-react-codemod` you can run `npm test` to check if everything still works.

Comment (iamdustan) [inline npm-react-codemod/test/pure-render-mixin-test2.output.js]: All of `npm-react-codemod`’s tests are failing on master for me right now.

```
Error: You have a collection of type [CallExpression]. "replaceWith" is is only defined for "Node"
```

Comment (iamdustan) [inline npm-react-codemod/test/pure-render-mixin-test2.output.js]: Additionally, I’m okay *not* changing these. I can back out these updates and only have docs updates. These tests will get a bit of noise due to the react/addons deprecation notice, but nothing will break at this time.

File: npm-react/README.md (hunk 1 above)
Comment (iamdustan) [inline npm-react/README.md]: added
```
// see http://facebook.github.io/react/docs/addons.html for a comprehensive list
```

File: npm-react-codemod/test/pure-render-mixin-test2.output.js (hunk 4 above)
Comment (cpojer) [inline npm-react-codemod/test/pure-render-mixin-test2.output.js]: Fixing up the tests; must have been the jscodeshift update by @fkling that made things a little more strict.

The code in these tests is not run. It verifies whether the transform works correctly. The purpose of this is to transform React.createClass calls to ES6 classes and this transform is one part of it. If we are suggesting people to use a different way to require react-addons we should update this codemod to also spit out code like this.

Comment (cpojer) [inline npm-react-codemod/test/pure-render-mixin-test2.output.js]: Fixed here: https://github.com/facebook/react/pull/3821

**Comment by zpao** [comment]
We ended up shipping the addons differently so closing this out.

**Pull request closed without merging**